src/advanced_inv.cpp

```diff
@@ -1752,15 +1753,18 @@ bool advanced_inventory::remove_item( advanced_inv_listitem &sitem )
         assert( &cont->contents.front() == sitem.it );
         cont->contents.erase( cont->contents.begin() );
         rc = true;
-    } else if( s.can_store_in_vehicle() && panes[src].in_vehicle() ) {
-        rc = s.veh->remove_item( s.vstor, sitem.it );
     } else if( sitem.area == AIM_WORN ) {
         rc = g->u.takeoff( sitem.it );
-    } else {
+    } else if( s.can_store_in_vehicle() ) {
+        rc = s.veh->remove_item( s.vstor, sitem.it );
+    }
+
+    if( !rc ) {
         g->m.i_rem( s.pos, sitem.it );
-        rc = true;
+        rc = true; // NOT ALWAYS! FIX!
     }
-    return rc;
+
+    return rc; // Always returns true
```

Comment:
Why do you want those function to return a value? What are you going to do with it? Besides displaying a debug message, which can happen already be done in the specific remove function.